src/view/com/post-thread/PostThread.tsx

```diff
@@ -215,24 +254,52 @@ function PostThreadLoaded({
           offset: pageY,
         })
       }
-      if (isNative) {
-        highlightedPostRef.current?.measure(
-          (_x, _y, _width, _height, _pageX, pageY) => {
-            onMeasure(pageY)
-          },
-        )
-      } else {
-        // Measure synchronously to avoid a layout jump.
-        const domNode = highlightedPostRef.current
-        if (domNode) {
-          const pageY = (domNode as any as Element).getBoundingClientRect().top
-          onMeasure(pageY)
-        }
+      // Measure synchronously to avoid a layout jump.
+      const domNode = highlightedPostRef.current
+      if (domNode) {
+        // @ts-ignore web only
+        const pageY = (domNode as any as Element).getBoundingClientRect().top
+        onMeasure(pageY)
       }
       needsScrollAdjustment.current = false
     }
   }, [thread])
 
+  // HACK
+  // Until we can get maintainVisibleContentPosition to work regardless of how many posts we have rendered at the
+  // top, this will simulate a "load" when we reach the top, to allow for the next items to render without jumping
+  const onStartReached = React.useCallback(() => {
+    // Get the first post. We need to get the second item in the array if the first one is the TOP_COMPONENT
+    let first = posts?.[0]
+    if (first === TOP_COMPONENT) {
+      first = posts?.[1]
+    }
+
+    // We do nothing in these situations
+    // - We are already prepending
+    // - This is not a thread post
+    // - There are no more parents to show
+    // - The top post is the highlighted post
+    if (
+      isPrepending.current ||
+      !isThreadPost(first) ||
+      !first.parent ||
+      first.ctx.isHighlightedPost
+    ) {
+      return
+    }
+
+    // Start prepending
+    isPrepending.current = true
+    // We wait a moment both to appear like a "load" event and to let the scroll "settle". 850ms is the sweet spot
+    // that we should be confident the scroll has settled at. Lower is slightly janky, higher is too long.
+    setTimeout(() => {
+      // Increment the top page count and set prepending to false
+      setTopPageCount(prev => prev + 1)
+      isPrepending.current = false
+    }, 850)
+  }, [posts])
```

Comment:
Because the animation needs to settle, the 850 ms does that. I'm going to see if we can instead use `onMomentumScrollEnd` to run the `setState()` so we can ensure reliability.

Reply:
Oh, I stand corrected. There's an `onScrollAnimationEnd` that I think is even better. Checking now.

Reply:
Oh, weird. There's a typehint for `onScrollAnimationEnd` but it isn't documented (and doesn't do anything). Is this something we added?

Anyway, `onMomentumScrollEnd` does what we want.